editoast/src/views/timetable.rs

```diff
@@ -750,6 +751,170 @@ fn build_conflict_core_request(
+    let previous = if current <= 1 {
+        None
+    } else {
+        Some(current - 1)
+    };
+    let next = if current >= page_size {
+        None
+    } else {
+        Some(current + 1)
+    };
+    let trains_count = Timetable::trains_count(timetable_id, conn).await? as u64;
+    let paced_trains_count = Timetable::paced_trains_count(timetable_id, conn).await? as u64;
+    let count = trains_count + paced_trains_count;
+    let nb_train_pages = (trains_count + page_size - 1) / page_size;
+    let nb_paced_train_pages = (paced_trains_count + page_size - 1) / page_size;
+    let page_count = nb_train_pages + nb_paced_train_pages;
+    let page_stats = PaginationStats {
+        count,
+        page_size,
+        page_count,
+        current,
+        previous,
+        next,
+    };
+
+    let (start_times, simulations) = if current <= nb_train_pages {
+        // Query concerns train schedules here
+        let ts_settings = page_settings
+            .into_selection_settings()
+            .filter(move || models::TrainSchedule::TIMETABLE_ID.eq(timetable_id));
+        let (trains, _) = models::TrainSchedule::list_paginated(conn, ts_settings).await?;
+        let simulations: Vec<_> = train_simulation_batch(
+            conn,
+            valkey_client.clone(),
+            core_client.clone(),
+            &trains,
+            &infra,
+            electrical_profile_set_id,
+        )
+        .await?
+        .into_iter()
+        .map(|(sim, _)| sim)
+        .collect();
+        let start_times = trains.iter().map(|ts| ts.start_time()).collect();
+        (start_times, simulations)
+    } else {
+        // Query concerns paced trains here: page queryed should be page_settings.page - nb_train_pages
+        page_settings.page = current - nb_train_pages;
+        let pt_settings = page_settings
+            .into_selection_settings()
+            .filter(move || models::PacedTrain::TIMETABLE_ID.eq(timetable_id));
+        let (paced_trains, _) = models::PacedTrain::list_paginated(conn, pt_settings).await?;
+        let (_, occurence_trains): (Vec<_>, Vec<_>) = paced_trains
+            .iter()
+            .flat_map(|pt| pt.iter_occurrences())
+            .unzip();
+        let simulations: Vec<_> = train_simulation_batch(
+            conn,
+            valkey_client.clone(),
+            core_client.clone(),
+            &occurence_trains,
+            &infra,
+            electrical_profile_set_id,
+        )
+        .await?
+        .into_iter()
+        .map(|(sim, _)| sim)
+        .collect();
+        let start_times = occurence_trains.iter().map(|ts| ts.start_time()).collect();
+        (start_times, simulations)
+    };
+    Ok(Json(TrainRequirementsCountList {
+        results: build_trains_requirements(start_times, simulations),
+        stats: page_stats,
+    }))
+}
+
+fn build_trains_requirements(
+    start_times: Vec<DateTime<Utc>>,
+    simulations: Vec<Arc<simulation::Response>>,
```

Comment:
```suggestion
    start_times: impl Iterator<Item = DateTime<Utc>>,
    simulations: impl Iterator<Item = simulation::Response>,
```

When random access isn’t needed, taking slices or vectors is usually unnecessary.

We don’t need to take an arc just to unwrap it. We know we can unwrap it because `train_simulation_batch` works that way. So, maybe better to unwrap it just after it returns?